src/test/kotlin/app/batch/MessageProducerTest.kt

```diff
@@ -0,0 +1,273 @@
+package app.batch
+
+import app.domain.DataKeyResult
+import app.domain.EncryptionResult
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.core.Appender
+import com.beust.klaxon.JsonObject
+import com.beust.klaxon.Parser
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import junit.framework.Assert.assertEquals
+import org.everit.json.schema.loader.SchemaLoader
+import org.json.JSONObject
+import org.json.JSONTokener
+import org.junit.Test
+import org.slf4j.LoggerFactory
+
+class MessageProducerTest {
```

Comment:
Although it's not a specific AC, I think we need a test that bad json doesn't blow up the app and is handled (i.e. rejected). It's important we can handle bad json files given the data structures we've seen in preprod so far.

Reply:
Yes we do

Reply:
This is tested - see should_Log_And_Continue_When_DBObject_IsNot_Valid_Json in HBaseWriterTest.